lib/BoundaryValueDiffEqMIRK/src/sparse_jacobians.jl

```diff
@@ -10,26 +10,33 @@ coloring.
 If the problem is a TwoPointBVProblem, then this is the complete Jacobian, else it only
 computes the sparse part excluding the contributions from the boundary conditions.
 """
-function __generate_sparse_jacobian_prototype(cache::MIRKCache, ya, yb, M, N)
-    return __generate_sparse_jacobian_prototype(cache, cache.problem_type, ya, yb, M, N)
+function __generate_sparse_jacobian_prototype(cache::MIRKCache, ya, yb, M, N, ad)
+    return __generate_sparse_jacobian_prototype(cache, cache.problem_type, ya, yb, M, N, ad)
 end
 
 function __generate_sparse_jacobian_prototype(
-        ::MIRKCache, ::StandardBVProblem, ya, yb, M, N)
+        ::MIRKCache, ::StandardBVProblem, ya, yb, M, N, ad)
     fast_scalar_indexing(ya) ||
         error("Sparse Jacobians are only supported for Fast Scalar Index-able Arrays")
     J_c = BandedMatrix(Ones{eltype(ya)}(M * (N - 1), M * N), (1, 2M - 1))
-    return ColoredMatrix(J_c, matrix_colors(J_c'), matrix_colors(J_c))
+    problem = ColoringProblem(;
+        partition = ifelse((ADTypes.mode(ad) isa ADTypes.ReverseMode), :row, :column))
+    algo = GreedyColoringAlgorithm()
+    result = coloring(J_c, problem, algo)
+    return result
 end
 
 function __generate_sparse_jacobian_prototype(
-        ::MIRKCache, ::TwoPointBVProblem, ya, yb, M, N)
+        ::MIRKCache, ::TwoPointBVProblem, ya, yb, M, N, ad)
     fast_scalar_indexing(ya) ||
         error("Sparse Jacobians are only supported for Fast Scalar Index-able Arrays")
     J₁ = length(ya) + length(yb) + M * (N - 1)
     J₂ = M * N
     J = BandedMatrix(Ones{eltype(ya)}(J₁, J₂), (M + 1, M + 1))
+    problem = ColoringProblem(;
+        partition = ifelse((ADTypes.mode(ad) isa ADTypes.ReverseMode), :row, :column))
+    algo = GreedyColoringAlgorithm()
     # for underdetermined systems we don't have banded qr implemented. use sparse
-    J₁ < J₂ && return ColoredMatrix(sparse(J), matrix_colors(J'), matrix_colors(J))
-    return ColoredMatrix(J, matrix_colors(J'), matrix_colors(J))
+    J₁ < J₂ && return coloring(sparse(J), problem, algo)
```

Comment:
The function `coloring` converts the matrix to `SparseMatrixCSC` anyway, so the `sparse` here is superfluous unless you need to recover a `SparseMatrixCSC` when you do `sparsity_pattern(coloring_result)` (aka `coloring_result.A`)